cico_setup.sh

```diff
@@ -102,26 +102,32 @@ function run_tests_with_coverage() {
 }
 
 function tag_push() {
-  TARGET=$1
-  docker tag fabric8-notification-deploy $TARGET
-  docker push $TARGET
+  local tag=$1
+
+  docker tag fabric8-notification-deploy $tag
+  docker push $tag
 }
 
 function deploy() {
+  if [ -n "${DEVSHIFT_USERNAME}" -a -n "${DEVSHIFT_PASSWORD}" ]; then
+    docker login -u ${DEVSHIFT_USERNAME} -p ${DEVSHIFT_PASSWORD} ${REGISTRY}
+  else
+    echo "Could not login, missing credentials for the registry"
```

Comment:
can we be explicit about the registry that we are looking creds for? Or will it obvious from some other place?

Reply:
maybe we could remove that `if` and force the docker login, as the registry is hardcoded... Is that what you mean?

Reply:
I don't think it's necessary though, if the `docker login` is not executed, the `docker build` and `docker push` statements will fail verbosely indicating why, I think.

Reply:
In my mind I was thinking since it a error statement execution will stop there, but then now I can see it is just an echo and with no exit statement.